proxy/src/dns.rs

```diff
@@ -121,11 +121,14 @@ impl Resolver {
 
     // `ResolverFuture` can only be used for one lookup, so we have to clone all
     // the state during each resolution.
-    fn lookup_ip(self, &Name(ref name): &Name) -> LookupIpFuture {
+    fn lookup_ip(self, &Name(ref name): &Name)
+        // TODO: unbox me
+        -> Box<Future<Item = LookupIp, Error = ResolveError>> {
+        let name = name.clone();
         let resolver = ResolverFuture::new(
```

Comment:
This function was changed in `trust-dns-resolver` to return `Box<Future<Item=ResolverFuture, ...>>`, which makes our lives a bit trickier here